### PR TITLE
[SKILL-BUTTON]: rewrite to put logic into component js

### LINKED ISSUE #1044: Rewrite skill-button so there's no unnecessarily deep HTML
# Problem

Right now, the `skill-button` is a `div` which, internally, renders a single, albeit complicated `button` tag.

We should rewrite it so that component itself is a `button` and all the associated bindings are determined at the component root level

## Subtasks
- [x] Remove `button` "wrapper" tag from the `skill-button.hbs
- [x] Add `tagName: button` to `skill-button.js`
- [x] Reimplement class and action binding at the `.js` instead of the `.hbs` level

This task probably requires a decent amount of ember component knowledge, but with guidance, should not be too complicated.

### PR DESCRIPTION
**Pull request by snewcomer**
# What's in this PR?
Move hbs logic into js file

## References
Fixes #1044

Progress on: #

### CODE REVIEW

File: app/components/skill-button.js [hunk 1]
```diff
@@ -15,8 +15,26 @@ export default Component.extend({
...
+  mouseEnter() {
+    set(this, 'isHovering', true);
+  },
+  mouseLeave() {
+    set(this, 'isHovering', false);
+  },
+
+  click() {
+    get(this, 'remove')();
+  },
```
Comment (JoshSmith) [inline app/components/skill-button.js]: Can you reorder the fns to be alpha-ordered and with a space between each?

File: app/components/skill-button.js [hunk 2]
```diff
@@ -15,8 +15,26 @@ export default Component.extend({
   isHovering: false,
   size: 'large',
 
+  classNames: ['has-spinner', 'skill'],
+  classNameBindings: ['isHovering:can-delete', 'size', 'type'],
+
+  attributeBindings: ['isLoading:disabled'],
+
+  tagName: 'button',
```
Comment (JoshSmith) [inline app/components/skill-button.js]: Can you interleave these with the top set of attributes in alpha order?

Comment (snewcomer) [inline app/components/skill-button.js]: fixed

**Comment by JoshSmith** [comment]
@snewcomer can you squash and merge?

**Pull request merged**